lms/views/api/canvas/files.py

```diff
@@ -29,7 +29,9 @@ def via_url(self):
             This exception is caught and handled by an exception view.
         """
         public_url = self.canvas_api_client.public_url(
-            self.request.matchdict["file_id"]
+            self.request.matchdict["course_id"],
+            self.request.matchdict["file_id"],
+            self.request.lti_user.is_instructor,
```

Comment:
Possibly worth using kwargs here to avoid any risk of a mismatch between the call and the definition.